gcc/rust/parse/rust-parse-impl.h

```diff
@@ -3139,11 +3139,17 @@ Parser<ManagedTokenSource>::parse_generic_params (EndTokenPred is_end_token)
 	break;
     }
 
-  // FIXME: Add reordering hint
   if (order_error)
-    rust_error_at (generic_params.front ()->get_locus (),
-		   "invalid order for generic parameters: lifetimes should "
-		   "always come before types");
+    {
+      rich_location rich_locus (line_table,
+				generic_params.front ()->get_locus ());
+      rich_locus.add_fixit_replace (
+	generic_params.front ()->get_locus (),
+	"reorder the parameters: lifetimes, then consts and types");
+      rust_error_at (
```

Comment:
We can't do a rust_error_at like this in the parser. You need to write a wrapper for a an error like this with the Error class you have worked with

Reply:
@philberty, currently we cannot bind `rich_location` to these parser errors. See this
- https://github.com/Rust-GCC/gccrs/issues/2566

So, I only added the updated the wrapper function, but didn't add the `fixit` hint :(